Gem/Code/Source/Automation/ScriptManager.cpp

```diff
@@ -1458,11 +1457,12 @@ namespace AtomSampleViewer
             // Note this will pause the script until the capture is complete
             if (PrepareForScreenCapture(filePath, s_instance->m_envPath))
             {
-                AZ_Assert(s_instance->m_frameCaptureId == AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId, "Attempting to start a capture while one is in progress");
-                uint32_t frameCaptureId = AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId;
+                AZ_Assert(s_instance->m_frameCaptureId == AZ::Render::InvalidFrameCaptureId, "Attempting to start a capture while one is in progress");
+                AZ::Render::FrameCaptureId frameCaptureId = AZ::Render::InvalidFrameCaptureId;
                 AZ::Render::FrameCaptureRequestBus::BroadcastResult(frameCaptureId, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, filePath);
-                if (frameCaptureId != AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId)
+                if (frameCaptureId != AZ::Render::InvalidFrameCaptureId)
```

Comment:
If this is false because the capture failed for some reason, we need to reset m_isCapturePending and unpause the script, to reverse what was done in PrepareForScreenCapture. Or move m_isCapturePending=true and PauseScript out of PrepareForScreenCapture.

Reply:
Updated to resume the script and log an error if the frame capture request fails here